### PR TITLE
Async web3 docs

### PR DESCRIPTION
**Pull request by kclowes**
### What was wrong?
This PR builds on #2819.

### How was it fixed?
Added some docs for the AsyncHTTPProvider!

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: docs/overview.rst [hunk 1]
```diff
@@ -13,12 +13,13 @@ greater detail.
...
-After installing web3.py (via ``pip install web3``), you'll need to specify the
-provider and any middleware you want to use beyond the defaults.
+After installing web3.py (via ``pip install web3``), you'll need to specify
+async or sync web3, the provider, and any middleware you want to use
+beyond the defaults.
 
 
-Providers
----------
+Sync Providers
```
Comment (pacrob) [inline docs/overview.rst:21]: I'd like this to be consistent with `AsyncProvider` below, but `SyncProviders` doesn't seem right. Maybe `Synchronous Providers` and `Asynchronous Providers`? Possibly a `Providers` header with a brief description of the differences?

File: docs/overview.rst [hunk 2]
```diff
@@ -47,6 +48,31 @@ For more information, (e.g., connecting to remote nodes, provider auto-detection
...
+   middleware can be seen on the :ref:`AsyncHTTPProvider` docs
+
+.. code-block:: python
+
+   >>> from web3 import AsyncWeb3
+
+   # AsyncHTTPProvider:
+   >>> w3 = AsyncWeb3(AsyncWeb3.AsyncHTTPProvider('http://127.0.0.1:8545'))
+
+   >>> w3.is_connected()
```
Comment (pacrob) [inline docs/overview.rst:69]:
```suggestion
   >>> await w3.is_connected()
```

Comment (pacrob) [reply docs/overview.rst:69]: `await` works if your in ipython. If it's just supposed to be the repl, then need to `import asyncio` and make it:
```suggestion
   >>> asyncio.run (w3.is_connected())
```

File: docs/quickstart.rst [hunk 3]
```diff
@@ -59,28 +59,36 @@ Local Providers
...
 
    # IPCProvider:
    >>> w3 = Web3(Web3.IPCProvider('./path/to/geth.ipc'))
 
    # HTTPProvider:
    >>> w3 = Web3(Web3.HTTPProvider('http://127.0.0.1:8545'))
 
+   # AsyncHTTPProvider:
+   >>> w3 = AsyncWeb3(AsyncWeb3.AsyncHTTPProvider('http://127.0.0.1:8545'))
+
```
Comment (pacrob) [inline docs/quickstart.rst:80]: same `await` or `asyncio.run`  for an `is_connected` here.

File: docs/overview.rst (hunk 2 above)
Comment (kclowes) [reply docs/overview.rst]: Nice catch! There is a box that says it's meant to be run in a python interpreter like ipython, so went with just `await`.

**Review by pacrob — APPROVED** [review]
lgtm!

**Review by fselmo — APPROVED** [review]
lgtm 👍🏼 Thanks!

**Pull request merged**